phys/module_ra_eclipse.F

```diff
@@ -0,0 +1,439 @@
+	zeta1  = sin_lat1 * sin(rd2) * E + cos_lat1 * cos(rd2) * cos(H)
+
+	!-- Correct cy (sembla que no és necessari)
+	cy1  = cy/rho1
+
+        !-- Distance to the shadow axis
+        DELTA2 = (xi - cx)**2 + (eta1  - cy1)**2;
+	DELTA  = sqrt(DELTA2)
+
+        !-- Penumbra and Umbra cone's circumference over the Earth's surface
+	LL1    = (cl1 - zeta1 * tanf1) ! Penumbra
+	LL2    = (cl2 - zeta1 * tanf2) ! Umbra
+
+        ! Check if we are inside the eclipse
+        if(DELTA .gt. LL1) then ! Without obscuration
+              Obscur(i,j) = 0.
+              mask(i,j)   = 0
+        elseif(DELTA .le. LL1 .and. DELTA .gt. abs(LL2)) then   ! Penumbra
+              Obscur(i,j) = (LL1 - DELTA) / (LL1 + LL2)         ! Partial eclipse
+              mask(i,j)   = 1
+        elseif(DELTA .le. abs(LL2)) then                        ! Umbra
+		if(LL2 .lt. 0) then
+                      Obscur(i,j) = 1.                          ! Total eclipse
+	              mask(i,j)   = 2
+                else
+                      Obscur(i,j) = (LL1 - DELTA) / (LL1 + LL2) ! Annular eclipse
+	              mask(i,j)   = 3
+                endif
+        endif
+
+        ! Check that the obscuration is not greater than 1
+        if(Obscur(i,j) .gt. 1) then
+              Obscur(i,j) = 1.
+        endif
+
+     enddo
+     ! End loop over longitudes
+  enddo
+! End loop over longitudes
+end subroutine
+!
+!----------------------------
+! INTERNAL SUBROUTINES
+!	load_besselian_elements --> load the besselian elements from eclipse_besselian_elements.dat
+!       compute_besselian_t     --> compute the besselian elements for the current time
+!
+!----------------------------
+subroutine load_besselian_elements(t, julday, year, X, Y, d, l1, l2, mu,  & 
+                                   tanf1, tanf2, t0, tmin, tmax, NBessel, &
+                                   eclipse_exist,Dt)
+  IMPLICIT NONE
+!
+!--------------
+!----------
+! VARIABLES
+!----------
+!--------------
+!
+!----------
+! Input/output vars
+!----------
+  integer,                             intent(in)    :: julday, year, NBessel
+  real,                                intent(in)    :: t
+  logical,                             intent(inout) :: eclipse_exist
+  real, dimension(NBessel),            intent(inout) :: X, Y, d, l1, l2, mu
+  real,                                intent(inout) :: tanf1, tanf2, t0, tmin, tmax, Dt
+!----------
+! Local vars
+!----------
+  integer :: ryear, rjulday
+  logical :: file_exist
+
+!----------
+! Initialization
+!----------
+  eclipse_exist = .false.
+
+!----------
+! Load the information from eclipse_besselian_elements.dat
+!----------
+  inquire( file="eclipse_besselian_elements.dat", exist=file_exist)
+  if (file_exist) then
+	 open (20,FILE='eclipse_besselian_elements.dat',status="old",action="read")
+  else
+	call wrf_error_fatal('load_besselian_elements: eclipse_besselian_elements.dat not found')
+	return
+  end if
+
+  do
```

Comment:
@davegill This is probably not standard for reading in data, right? How did it work for MPI runs?